.github/workflows/snapshot.yml

```diff
@@ -13,10 +13,27 @@ on:
         type: boolean
 
 jobs:
+  validate:
+    runs-on: ubuntu-latest
+    outputs:
+      is-snapshot: ${{ steps.validate.outputs.is-snapshot }}
+    steps:
+      - uses: actions/checkout@v4
+      - name: Validate version is a snapshot version
+        id: validate
+        run: |
+          output=false
+          if [[ "$(./mvnw -q help:evaluate -Dexpression=project.version -DforceStdout)" =~ "-SNAPSHOT" ]]; then
+            echo "This is a snapshot version"
+            output=true
+          fi
```

Comment:
What is the advantage of not failing over setting an output and skipping?

Reply:
Otherwise, teams will need to investigate whether it failed for that reason rather than something else.